streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -728,33 +745,57 @@ int maybeCommitActiveTasksPerUserRequested() {
-                    if (!offsetAndMetadata.isEmpty()) {
-                        consumedOffsetsAndMetadataPerTask.put(task.id(), offsetAndMetadata);
+            if (eosBetaEnabled && !eosUpgradeModeEnabled) {
+                for (final Task task : activeTaskIterable()) {
+                    if (task.commitRequested() && task.commitNeeded()) {
+                        return commitAll();
                     }
                 }
-            }
 
-            commitOffsetsOrTransaction(consumedOffsetsAndMetadataPerTask);
+                return 0;
+            } else {
+                final Map<TaskId, Map<TopicPartition, OffsetAndMetadata>> consumedOffsetsAndMetadataPerTask = new HashMap<>();
+
+                for (final Task task : activeTaskIterable()) {
+                    if (task.commitRequested() && task.commitNeeded()) {
+                        task.prepareCommit();
+                        final Map<TopicPartition, OffsetAndMetadata> offsetAndMetadata = task.committableOffsetsAndMetadata();
+                        if (!offsetAndMetadata.isEmpty()) {
+                            consumedOffsetsAndMetadataPerTask.put(task.id(), offsetAndMetadata);
+                        }
+                    }
+                }
 
-            for (final Task task : tasks.values()) {
-                if (consumedOffsetsAndMetadataPerTask.containsKey(task.id())) {
-                    task.postCommit();
+                commitOffsetsOrTransaction(consumedOffsetsAndMetadataPerTask);
+
+                for (final Task task : tasks.values()) {
+                    if (consumedOffsetsAndMetadataPerTask.containsKey(task.id())) {
+                        task.postCommit();
+                    }
                 }
-            }
 
-            return consumedOffsetsAndMetadataPerTask.size();
+                return consumedOffsetsAndMetadataPerTask.size();
+            }
         }
     }
 
     private void commitOffsetsOrTransaction(final Map<TaskId, Map<TopicPartition, OffsetAndMetadata>> offsetsPerTask) {
-        if (eosEnabled) {
+        if (eosAlphaEnabled) {
             for (final Map.Entry<TaskId, Map<TopicPartition, OffsetAndMetadata>> taskToCommit : offsetsPerTask.entrySet()) {
-                activeTaskCreator.streamsProducerForTask(taskToCommit.getKey()).commitTransaction(taskToCommit.getValue());
+                activeTaskCreator.streamsProducerForTask(taskToCommit.getKey())
+                    .commitTransaction(taskToCommit.getValue(), mainConsumer.groupMetadata().groupId());
+            }
+        } else if (eosBetaEnabled) {
+            if (eosUpgradeModeEnabled) {
+                for (final Map.Entry<TaskId, Map<TopicPartition, OffsetAndMetadata>> taskToCommit : offsetsPerTask.entrySet()) {
+                    activeTaskCreator.streamsProducerForTask(taskToCommit.getKey())
+                        .commitTransaction(taskToCommit.getValue(), mainConsumer.groupMetadata());
```

Comment:
Eos-beta during upgrade: we just pass `ConsumerGroupMetadata` during per-task tx-commit to enable consumer side "fetch offset blocking"